tests/unit/test_charm.py

```diff
@@ -384,3 +394,24 @@ def test_given_self_signed_certificates_not_yet_stored_when_on_certificate_creat
             WaitingStatus("Root Certificates are not yet set"),
             self.harness.charm.unit.status,
         )
+
+    def test_given_self_signed_certificates_with_additional_extensions(self):
+        """Test setting an extension on a CSR and retrieving it from the cert."""
+        peer_relation_id = self.harness.add_relation("replicas", self.harness.charm.app.name)
+        self.harness.add_relation_unit(peer_relation_id, self.harness.charm.unit.name)
+
+        csr = generate_csr(
+            generate_private_key(),
+            subject="my_subject",
+            sans=["www.canonical.com"],
```

Comment:
`www.canonical.com` should be a variable (ex. `san_0`). This way in the assert at the end, we don't validate that the SAN is specifically `www.canonical.com` but the value of `san_0`.